src/ui/modules/RichtextModule/TableOfContents.tsx

```
@@ -0,0 +1,78 @@
+            }
+          });
+        },
+        {
+          rootMargin: `-${headerHeight}px 0px 0px 0px`,
+        }
+      );
+
+      observer.observe(target);
+
+      return () => observer.disconnect();
+    });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)